syntax = "proto3";

package news_getter;

service NewsGetter {
  rpc SearchNews(SearchNewsRequest) returns (SearchNewsResponse);
  rpc GetNewsContent(GetNewsContentRequest) returns (GetNewsContentResponse);
}

message SearchNewsRequest {
  string query = 1;
  repeated Site sites = 2;
  int32 max_site_links = 3;
}

message SearchNewsResponse {
  repeated string links = 1;
}

message GetNewsContentRequest {
  string url = 1;
  Site scraper = 2;
}

message GetNewsContentResponse {
  bool is_success = 1;
  string url = 2;
  string title = 3;
  string content = 4;
  repeated string images = 5;
  repeated string videos = 6;
}

enum Site {
  AutoDetect = 0;
  LentaRu = 1;
  IzRu = 2;
  Ria = 3;
  Interfax = 4;
  Tass = 5;
  BBC = 6;
  CNN = 7;
}
